plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -804,9 +806,9 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 	 * @return array
 	 */
 	public function get_item_schema() {
-		$weight_unit    = get_option( 'woocommerce_weight_unit' );
-		$dimension_unit = get_option( 'woocommerce_dimension_unit' );
-		$schema         = array(
+		$weight_unit_label    = I18nUtil::get_weight_unit_label( get_option( 'woocommerce_weight_unit', 'kg' ) );
+		$dimension_unit_label = I18nUtil::get_dimensions_unit_label( get_option( 'woocommerce_dimension_unit', 'cm' ) );
+		$schema               = array(
```

Comment:
Back-compat question (not necessarily a blocker, but wanted to pose the question): supposing a merchant is unhappy with the default weight units as supported by WooCommerce (`kg` | `g` | `lbs` | `oz`) and prefers to use something appropriate to their use case, like the [`dwt` pennyweight](https://en.wikipedia.org/wiki/Pennyweight) and, to do this, they update the `woocommerce_weight_unit` option directly.

Is there a risk of creating a minor breakage in that, previously, this would work:

```php
get_option( 'woocommerce_weight_unit' ); # => 'dwt'
```

Whereas now, I think they would get back an empty string (at least, if I'm reading [this code](https://github.com/woocommerce/woocommerce/pull/36708/files#diff-e1d2d755cf727d32dc1c0e14d667f66e96201ce68cf78a2f6b3973e7b93ee1cbR31-R37) correctly)?

Reply:
Good point. Perhaps it would be better for the label to default to the unit value ("dwt" in this case) instead of an empty string?

Reply:
I think that might be the best course of action.

(It crossed my mind that we could additionally make the result filterable: but perhaps it is better to hold on that until a concrete request for such a thing surfaces—rather than prematurely increase the public API surface.)

Reply:
Updated in b336652e0d46bfa96c1257e5bd2424e25e07d632